### ISSUE TITLE
Fix connection issues with the Hana extension

### ISSUE THREAD

**Issue by parrot9design** [balancednetwork/balanced-network-interface#1363]
If you have Hana active in your browser, it will connect to EVM chains without any prompting. 

If you use Hana with multiple wallets, every time you view a different wallet in Hana the EVM wallet connected to Balanced will change to match. I'm worried this is going to catch someone out eventually and they'll send money from the wrong wallet (I almost have on several occasions).

This also causes some issues when trying to make transactions from the auto-connected chain. For example, I was just trying to swap / transfer from Avalanche. It fails to approve USDC transfers (doesn't even trigger the wallet to sign anything), and if it doesn't need approval first, clicking the button in the confirmation modal won't do anything. I have plenty of AVAX for gas fees, but discovered I could resolve the issue by changing the active EVM chain, then going back to Avalanche. TLDR: there's an issue with the auto-connected chain not being properly connected.

**Comment by hetfly** [comment]
>TLDR: there's an issue with the auto-connected chain not being properly connected.

This was likely fixed in previous releases

>If you use Hana with multiple wallets

Hana has implemented API standards for EVM wallets, meaning the events emitted from the frontend are caught by active EVM compatible wallets. I might be misunderstanding the issue but I would say this is up to the user to have only one EVM wallet activated in the browser. They would be facing similar issues on other apps as well, if they were using multiple EVM wallets at the same time.

**Comment by FezBox** [comment]
Closing as no longer an issue
